acra/acra-in-depth/security-design/_index.md

```diff
@@ -2,3 +2,165 @@
+
+We plan to include more triggers into the future releases.
+
+Once the trigger is triggered, an automatic reaction can be executed. You can choose to either:
+- notify a 3rd party monitoring / alarm system or
+- stop data decryption.
+
+### Guarantee 3: Add encryption context
+
+By differentiating the sensitive data via [Zones]({{< ref "acra/acra-in-depth/cryptography-and-key-management/#zones-INVALID" >}}), Acra users can build an additional layer of security:
+- Zone keys are used for encrypting the AcraStruct's RK section. In this case, AcraServer will only be able to pick the right key and decrypt the data by knowing the correct zone. This limits the leak scope for every private key, even if AcraServer is partially compromised.
+- Zone IDs are used as context when encrypting Secure Cell, providing additional protection.
+
+Zone data is supplied to an app via JSON (i.e. `{"id": "...", "public_key": "..."}`) from [HTTP api]({{< ref "acra/configuring-maintaining/general-configuration/acra-server.md#http-api-INVALID">}}) or [console utility]({{< ref "acra/security-controls/transport-security/acra-connector.md#acraconnector-and-acrawriter-INVALID" >}}).
+
+
+## Security model for Acra
+
+### Possible threats
+
+> Note: We recommend taking a look at the [architectural scheme of Acra]({{< ref "acra/acra-in-depth/data-flow/#-INVALID" >}}) before continuing to read.
+
+Acra at its core is a set of tools that allow safeguarding the security of a database (running PostgreSQL or MySQL) against the known widespread threats.
+
+The most dangerous current known security threats are:
+- Threats related to excess / abuse of access privileges.
+- Data leaks caused by mistakes in deploy, configuration settings, backup thefts, etc.
+- SQL injections.
+- Denial of service.
+- Vulnerabilities in the database protocol.
+- Weak audit or an absence thereof.
+- Operational system vulnerabilities.
+- Unsafe handling of cryptographic keys.
+
+### Conditions of secure work
+Acra can perform its protective functions properly and protect from the security threats 1, 2, 3 if the following security assumptions are met:
+- The PKI infrastructure is trusted;
+- AcraServer is trusted;
+- The client is less trusted than the server.
+
+### Possible consequences of compromisation
+Let’s consider all the possible consequences of any of separate component being broken (broken as in “fully compromised” when the adversary fully overtakes the work of the component and gains full access to its memory).
+
+When a *Database* is broken into, the worst-case scenario is DoS or COA. Thus, the stability of the system, in this case, is reduced to the stability of the symmetric encryption algorithm (AES-GCM-256).
+When the *Client* gets broken, the worst-case scenario is that the adversary can get the data belonging to this client, which is stored in the database.
+And finally, if AcraServer gets broken, the adversary can fully compromise the system.
```

Comment:
It would be better to define what “broken” means in this context. Data dump on the public internet? Leaked keys? Both? Another heartbleed found? Fundamental flaw in AES discovered? Attacker gaining control over the process? The server put out of service by DoS? The datacenter being on fire?

Also, I think COA needs to be expanded since it's not a common abbreviation. At least I had no clue what it means, [the Wikipedia list](https://en.wikipedia.org/wiki/COA) suggests that ciphertext-only attack might be it. I think this page would benefit from being more accessible to non-cryptographers.

Reply:
> It would be better to define what “broken” means in this context

actually, all of these things and we mention the worst of them on our opinion that related to data security.

> Also, I think COA needs to be expanded since it's not a common abbreviation.

agree, changed to `COA (ciphertext-only attack)`